### PR TITLE
Add missing THREAD_POOL_TYPES

### PR DESCRIPTION
**Pull request by prdoyle**
Add a couple of missing entries to `ThreadPool.THREAD_POOL_TYPES`.

There's probably more refactoring to do here. In production, this map is used only by `ExecutorNames` to ensure all names correspond to built-in thread pools; all other uses are in tests.

Furthermore, this map has a lot of overlap with the `builders` field, because the builder classes map 1:1 with the `ThreadPoolType` values. However, `THREAD_POOL_TYPES` is static while `builders` is not, so we can't derive the former from the latter.

### CODE REVIEW

**Comment by prdoyle** [comment]
Thanks for nothing @elasticsearchmachine

**Comment by prdoyle** [comment]
@elasticsearchmachine help

**Review by rjernst — CHANGES_REQUESTED** [review]
Adding the missing types seems fine, but it looks like there are some unrelated changes.

File: server/src/main/java/org/elasticsearch/indices/SystemIndexDescriptor.java [hunk 1]
```diff
@@ -340,18 +339,6 @@ protected SystemIndexDescriptor(
             }
         }
 
-        if (Objects.nonNull(executorNames)) {
```
[the remark this answers is not in the document]
Comment (rjernst) [reply server/src/main/java/org/elasticsearch/indices/SystemIndexDescriptor.java]: Got it, thanks. Makes sense then.

**Review by rjernst — APPROVED** [review]
LGTM

**Pull request merged**